src/Tools/Source/RunTests/ProcessUtil.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Management;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RunTests
+{
+    internal static class ProcessUtil
+    {
+        internal static int? GetParentProcessId(Process p)
+        {
+            try
+            {
+                ManagementObject mo = new ManagementObject("win32_process.handle='" + p.Id + "'");
+                mo.Get();
+                return Convert.ToInt32(mo["ParentProcessId"]);
+            }
+            catch 
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Return the list of processes which are direct children of the provided <paramref name="process"/> 
+        /// instance.
+        /// </summary>
+        /// <remarks>
+        /// This is a best effort API.  It can be thwarted by process instances starting / stopping during
+        /// the building of this list.
+        /// </remarks>
+        internal static List<Process> GetProcessChildren(Process process) => GetProcessChildrenCore(process, Process.GetProcesses());
+
+        private static List<Process> GetProcessChildrenCore(Process parentProcess, IEnumerable<Process> processes)
+        {
+            var list = new List<Process>();
+            foreach (var process in processes)
+            {
+                var parentId = GetParentProcessId(process);
+                if (parentId == parentProcess.Id)
+                {
+                    list.Add(process);
+                }
+            }
+
+            return list;
+        }
+
+        /// <summary>
+        /// Return the list of processes which are direct or indirect children of the provided <paramref name="process"/> 
+        /// instance.
+        /// </summary>
+        /// <remarks>
+        /// This is a best effort API.  It can be thwarted by process instances starting / stopping during
+        /// the building of this list.
+        /// </remarks>
+        internal static List<Process> GetProcessTree(Process process)
```

Comment:
Is this being used anywhere? Trying to rationalize why there's both this and GetProcessChildren.